hnn_core/parallel_backends.py

```diff
@@ -150,15 +161,17 @@ class MPIBackend(object):
 
     """
     def __init__(self, n_procs=None, mpi_cmd='mpiexec'):
-        self.n_procs = n_procs
         n_logical_cores = multiprocessing.cpu_count()
```

Comment:
Might be better to use [`psutil`](https://psutil.readthedocs.io/en/latest/#psutil.cpu_count) module? I've run into issues on computing clusters where `multiprocessing.cpu_count()` returns all cores on a node, rather than cores available to the user.

This could be something left to the user to define explicitly, but in any case the replacement would be:
```py
 n_logical_cores = len(psutil.Process().cpu_affinity())
```

Reply:
Keen observation. Yes, I've run into the same, which is why I use `os.sched_getaffinity(0)` a few lines below. I think both ways achieve the same end result.

Reply:
Thanks for reviewing @ntolley!